From 11a3f8dd1b1ad3f729332d105bfe3279b7628564 Mon Sep 17 00:00:00 2001
From: Jakub Palider <jpalider@marvell.com>
Date: Thu, 11 Feb 2021 03:53:17 +0530
Subject: [PATCH 730/767] misc: bphy: Add ioctl to get max irq number

commit c516399919a2b2295013aa8599241f65d025d90b from
git@git.assembla.com:cavium/WindRiver.linux.git

This patch aims at providing users of this driver with
maximum IRQ number via a ioctl call.

Signed-off-by: Jakub Palider <jpalider@marvell.com>
Change-Id: I9165517f5e056217f51cda43261810ac3e718de7
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/45891
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Radha Chintakuntla <radhac@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/47713
Reviewed-by: Naveen Mamindlapalli <Naveen.Mamindlapalli@cavium.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/misc/otx_bphy_ctr.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/misc/otx_bphy_ctr.c b/drivers/misc/otx_bphy_ctr.c
index 0bdb1616b7c8..029ed5d49570 100644
--- a/drivers/misc/otx_bphy_ctr.c
+++ b/drivers/misc/otx_bphy_ctr.c
@@ -52,6 +52,9 @@ struct otx_irq_usr_data {
 #define OTX_IOC_CLR_BPHY_HANDLER \
 	_IO(OTX_IOC_MAGIC, 2)
 
+#define OTX_IOC_GET_BPHY_MAX_IRQ \
+	_IOR(OTX_IOC_MAGIC, 3, u64)
+
 static inline int __install_el3_inthandler(unsigned long irq_num,
 					   unsigned long sp,
 					   unsigned long cpu,
@@ -154,6 +157,11 @@ static long otx_dev_ioctl(struct file *f, unsigned int cmd, unsigned long arg)
 		if (ret != 0)
 			return -ENOENT;
 		break;
+	case OTX_IOC_GET_BPHY_MAX_IRQ:
+		irq_num = MAX_IRQ;
+		if (copy_to_user((u64 *)arg, &irq_num, sizeof(irq_num)))
+			return -EFAULT;
+		break;
 	default:
 		return -ENOTTY;
 	}
-- 
2.31.1

